src/navigation/lib/initializeNavigationState.tsx

```diff
@@ -0,0 +1,104 @@
+import type { NavigationState, ViewRegistryItem, View, ContentState } from '@/types'
+import { ViewWrapper } from '@/components/ViewWrapper'
+import * as views from '@/views'
+import * as uuid from 'uuid'
+import { ViewProvider } from '@/contexts/ViewProvider'
+import { currentView } from '@/navigation/lib'
+import { calculateViewWidts } from '@/navigation/lib/calculateViews'
+
+const registeredComponents = new Map() as Map<string, ViewRegistryItem>
+
+export function initializeNavigationState(): NavigationState {
+  Object.keys(views).forEach((name) => {
+    registeredComponents.set(name, {
+      component: views[name as View],
+      meta: views[name as View].meta
+    })
+  })
+
+  const { name = 'start', props } = currentView()
+  if (!history?.state?.contentState?.length) {
+    const InititalView = viewRegistry.getByPath(window.location.pathname)
+    const id = uuid.v4()
+
+    history.replaceState({
+      id,
+      viewName: name,
+      contentState: [{
+        id,
+        name,
+        props,
+        path: window.location.pathname
+      }]
+    }, '', window.location.href)
+
+    return {
+      viewRegistry,
+      views: [{ name, colSpan: 12 }],
+      focus: null,
+      active: id,
+      content: [(
+        <ViewProvider key={id} id={id} name={name}>
+          <ViewWrapper colSpan={12}>
+            <InititalView.component id={id} />
+          </ViewWrapper>
+        </ViewProvider>
+      )]
+    }
+  }
+
+
+  const preContent = history.state.contentState.map((item: ContentState): { name: string } => {
+    return item
+  })
```

Comment:
```suggestion
  // Recreate navigationstate from history
  const preContent = history.state.contentState.map((item: ContentState): { name: string } => {
    return item
  })
```
Just for clarity, your choice!